esp/clients/ws_dfsclient/ws_dfsclient.cpp

```diff
@@ -623,8 +635,24 @@ IDFSFile *lookupDFSFile(const char *logicalName, unsigned timeoutSecs, unsigned
             throw makeStringExceptionV(-1, "eclwatch '%s': service port not defined", eclWatchName.str());
         const char *protocol = eclWatch.getPropBool("@tls") ? "https" : "http";
         serviceUrl.appendf("%s://%s:%u", protocol, result.first.c_str(), result.second);
-    }
+#else
+        static std::vector<std::string> dfsServiceUrls;
+        static CriticalSection dfsServiceUrlCrit;
+        static std::atomic<unsigned> currentDfsServiceUrl{0};
+        static std::atomic<bool> dfsServiceUrlsDiscovered{false};
+        bool expected = false;
+        if (dfsServiceUrlsDiscovered.compare_exchange_strong(expected, true))
```

Comment:
I'm not sure this is what you want.  It looks like you need a critical section - otherwise a second thread could execute lines 651 before the 1st thread executes 646,

Reply:
true, I'l change to crit pattern.